tests/base/src/test/java/org/wildfly/security/auth/server/CustomNameRewriterTest.java

```diff
@@ -0,0 +1,75 @@
+
+    private static final String BEFORE_USER_NAME = "bob";
+    private static final String AFTER_USER_NAME = "Robert";
+
+    @Test
+    public void testCustomNameRewriter() {
+        CustomNameRewriter rewriter = new CustomNameRewriter();
+        String adjustedName = rewriter.rewriteName(BEFORE_USER_NAME);
+        assertEquals(AFTER_USER_NAME, adjustedName);
+    }
+
+    @Test
+    public void testCustomNameRewriterAuthentication() throws Exception {
+        FileSystemSecurityRealm fileSystemSecurityRealm = createSecurityRealm();
+        CustomNameRewriter rewriter = new CustomNameRewriter();
+        SecurityDomain domainWithRewriter = SecurityDomain.builder().setDefaultRealmName("default").addRealm("default", fileSystemSecurityRealm).build()
+                .setPermissionMapper(((permissionMappable, roles) -> LoginPermission.getInstance()))
+                .setPreRealmRewriter(rewriter)
+                .build();
+        ServerAuthenticationContext sac1 = domainWithRewriter.createNewAuthenticationContext();
+        sac1.setAuthenticationName("bob"); // security domain contains the user "Robert"
+        assertTrue(sac1.authorize());
+
+        SecurityDomain domainWithoutRewriter = SecurityDomain.builder().setDefaultRealmName("default").addRealm("default", fileSystemSecurityRealm).build()
+                .setPermissionMapper(((permissionMappable, roles) -> LoginPermission.getInstance()))
+                .build();
+        ServerAuthenticationContext sac2 = domainWithoutRewriter.createNewAuthenticationContext();
+        sac2.setAuthenticationName("bob"); // should fail if rewriter not configured
+        assertFalse(sac2.authorize());
+    }
+
+    @Test
+    public void testCustomNameRewriterNonExistingUser() throws Exception{
+        FileSystemSecurityRealm fileSystemSecurityRealm = createSecurityRealm();
+        CustomNameRewriter rewriter = new CustomNameRewriter();
+        SecurityDomain securityDomain = SecurityDomain.builder().setDefaultRealmName("default").addRealm("default", fileSystemSecurityRealm).build()
+                .setPermissionMapper(((permissionMappable, roles) -> LoginPermission.getInstance()))
+                .setPreRealmRewriter(rewriter)
+                .build();
+        ServerAuthenticationContext sac = securityDomain.createNewAuthenticationContext();
+        sac.setAuthenticationName("john");
+        assertFalse(sac.authorize());
+    }
+
+    private FileSystemSecurityRealm createSecurityRealm() throws Exception {
+        FileSystemSecurityRealm realm = new FileSystemSecurityRealm(ServerUtils.getRootPath(true, getClass()));
+        ServerUtils.addUser(realm, "Robert");
+        return realm;
+    }
+
+    private final class CustomNameRewriter implements NameRewriter {
+
+        @Override
+        public String rewriteName(String original) {
+            if (original == null) {
+                return null;
+            } else if (original.equals("bob")) {
+                return "Robert";
```

Comment:
Thank you for the PR! 
Something minor: since you have "bob" and "Robert" Configured as constants, are you able to use those here instead?